aten/src/ATen/core/jit_type.h

```diff
@@ -531,17 +559,29 @@ struct CAFFE2_API TensorType : public Type {
 
   TensorTypePtr withDim(c10::optional<size_t> d) {
     auto copy = clone();
-    copy->sizes_ = VaryingShape(d);
-    copy->strides_ = VaryingShape(d);
+    //withDim is only used by the legacy executor
+    // that only cares about the rank, so create dummy symbols)) :
+    copy->sizes_ = d.has_value() ? VaryingShape<ShapeSymbol>(*d) : VaryingShape<ShapeSymbol>();
+    copy->strides_ = VaryingShape<int64_t>(d);
```

Comment:
this looks like it is missing correct handling of stride_indices_ and contiguity_. This is another reasons to implement strides a a self-contained object: a bunch of splatted parameters make it harder to keep objects up to date.